src/WebJobs.Extensions.IoTHub/Converters/JObjectToMessage.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.Devices;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.IoTHub.Converters
+{
+    internal class JObjectToMessage : IConverter<JObject, Message>
+    {
+        public Message Convert(JObject input)
+        {
+            JToken body = null;
+            var message = input.ToObject<Message>();
+
+            // by convention, use a 'body' property to initialize method
+            if (input.TryGetValue("body", StringComparison.OrdinalIgnoreCase, out body))
+            {
+                // can only set body through constructor
+                var messageWithBody = new Message(Encoding.UTF8.GetBytes((string)body));
+                messageWithBody.Ack = message.Ack;
```

Comment:
You may be able to use  Populate() to simplify this: 
    JsonConvert.PopulateObject(input.ToString(), msg);